src/rgw/CMakeLists.txt

```diff
@@ -9,6 +9,12 @@ if(WITH_RADOSGW_SELECT_PARQUET)
   message("-- arrow is installed, radosgw/s3select-op is able to process parquet objects")
 endif(WITH_RADOSGW_SELECT_PARQUET)
 
+if(WITH_RADOSGW_ARROW_FLIGHT)
+  set(ARROW_FLIGHT_LIBRARIES Arrow::Flight)
+  add_definitions(-D_ARROW_FLIGHT_EXIST)
```

Comment:
i'd avoid `add_definitions()` globally. how about setting `COMPILE_DEFINITIONS` property for the target which is interested in this macro? also, `add_compile_definitions()` is preferred over `add_definitions()` per cmake document in this use case.